crates/cdk-ffi/src/wallet.rs

```diff
@@ -0,0 +1,293 @@
+//! FFI Wallet bindings
+
+use std::sync::Arc;
+
+use cdk::wallet::{Wallet as CdkWallet, WalletBuilder as CdkWalletBuilder};
+use cdk_sqlite::wallet::{memory, WalletSqliteDatabase};
+
+use crate::error::FfiError;
+use crate::types::*;
+
+/// FFI-compatible Wallet
+#[derive(uniffi::Object)]
+pub struct Wallet {
+    inner: Arc<CdkWallet>,
+}
+
+#[uniffi::export]
+impl Wallet {
+    /// Create a new Wallet
+    #[uniffi::constructor]
+    pub async fn new(
+        mint_url: String,
+        unit: CurrencyUnit,
+        seed: Vec<u8>,
+        config: WalletConfig,
+    ) -> Result<Self, FfiError> {
+        let localstore: Arc<
+            dyn cdk_common::database::WalletDatabase<Err = cdk_common::database::Error>
+                + Send
+                + Sync,
+        > = if let Some(database_path) = &config.database_path {
+            Arc::new(
+                WalletSqliteDatabase::new(database_path.as_str())
+                    .await
+                    .map_err(|e| FfiError::Database { msg: e.to_string() })?,
+            )
+        } else {
+            Arc::new(
+                memory::empty()
+                    .await
+                    .map_err(|e| FfiError::Database { msg: e.to_string() })?,
+            )
+        };
```

Comment:
I don't think we should default to a memory db. I think it's safer to require a work_dir.